app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -307,102 +309,115 @@ public void resendConsentAgreement(Study study, SubpopulationGuid subpopGuid, St
         }).collect(BridgeCollectors.toImmutableList());
     }
 
-    public PagedResourceList<? extends UploadView> getUploads(Study study, String userId, DateTime startTime, DateTime endTime) {
+    public PagedResourceList<? extends UploadView> getUploads(Study study, String userId, DateTime startTime,
+            DateTime endTime) {
         checkNotNull(study);
         checkNotNull(userId);
-        
+
         Account account = getAccountThrowingException(study, userId);
-        
+
         return uploadService.getUploads(account.getHealthCode(), startTime, endTime);
     }
-    
+
     public List<NotificationRegistration> listRegistrations(Study study, String userId) {
         checkNotNull(study);
         checkNotNull(userId);
-        
+
         Account account = getAccountThrowingException(study, userId);
-        
+
         return notificationsService.listRegistrations(account.getHealthCode());
     }
-    
+
     public void sendNotification(Study study, String userId, NotificationMessage message) {
         checkNotNull(study);
         checkNotNull(userId);
         checkNotNull(message);
-        
+
         Account account = getAccountThrowingException(study, userId);
 
         notificationsService.sendNotificationToUser(study.getStudyIdentifier(), account.getHealthCode(), message);
     }
-    
+
     private IdentifierHolder saveParticipant(Study study, Set<Roles> callerRoles, StudyParticipant participant,
             boolean isNew, boolean sendVerifyEmail) {
         checkNotNull(study);
         checkNotNull(callerRoles);
         checkNotNull(participant);
-        
+
         Validate.entityThrowingException(new StudyParticipantValidator(study, isNew), participant);
+        
+        boolean newAccountAssigningId = isNew && study.isExternalIdValidationEnabled()
+                && isNotBlank(participant.getExternalId());
         Account account = null;
         if (isNew) {
             // Don't set it yet. Create the user first, and only assign it if that's successful.
             // Allows us to assure that credentials and ID will be related or not created at all.
-            if (isNotBlank(participant.getExternalId())) {
-                externalIdService.reserveExternalId(study, participant.getExternalId());    
+            if (newAccountAssigningId) {
+                externalIdService.reserveExternalId(study, participant.getExternalId());
```

Comment:
This check is now redundant with ExternalIdService. I would argue that having this check in ParticipantService is the cause of the complexity explosion we're seeing. We're making the code less modular and trying to make ParticipantService do too much.